src/test/recover.zig

```
@@ -0,0 +1,131 @@
+// Copyright © 2024 Dimitris Dinodimos.
+
+//! Panic recover.
+//! Regains control of the calling thread when the function panics or behaves
+//! undefined.
```

I'm skeptical of this. Recovering from panics isn't really intended to be possible or supported. IMO we should just let it crash, or if we really need fine-grained panic reporting in CI we could run each test in a new process using a filter.

---

IDk, I'd personally rather have my test suite finish running all tests instead of aborting early when one of them panics. Thoughts?

---

I'm concerned that panic recovery will fail or cause unexpected results, since this isn't really something Zig has put time into making possible:

- ziglang/zig#1356
- https://github.com/ziglang/zig/issues/1656#issuecomment-840190292 (recover.zig uses setjmp/longjmp or functions with the same issues)

So realistically it could mess up tests that run after the panic. But I guess that might not be terrible for a worst-case scenario since you will just see that fixing the panic causes the tests after it to work again.

---

I've been using this same panic recovery handler for testing in my side projects; I haven't had problems with it yet.